docs/sdkx-cocos2dx/notifications-android.mdx

```diff
@@ -0,0 +1,264 @@
+---
+sidebar_position: 50
+title: Notifications Android
+description: "Learn more about configuring push and in-app notifications."
+---
+
+import {
+  Admonition,
+  Centered,
+  Image,
+  Intro,
+} from "@site/src/components/forDocs";
+
+# Notifications Android {#notifications}
+
+<Intro>
+
+Configure Push notifications
+
+</Intro>
+
+<Admonition type="info" title="Note">
+
+All the public APIs in the SDK should be called after initializing the SDK via [HelpshiftCocosBridge.install() API](/sdkx-cocos2dx/getting-started#initializing)
+
+</Admonition>
+
+## Push notifications via Helpshift {#push-via-helpshift}
+
+<Image
+  src="/static/books/sdkx_android/app_notification.png"
+  width="half"
+  alt="urbanAirshipNotif.png"
+/>{" "}
+
+Helpshift allows you to send Push notifications when an agent replies to a conversation.
+
+### Prerequisites {#gcm-prerequisites}
+
+Implement FCM push in your app.
+
+For FCM, refer to the
+[Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging) documentation.
+
+### Configure Helpshift Agent Dashboard {#configure-helpshift-push-admin}
+
+To enable the Helpshift system to send push notifications to your users
+you will have to add an **Android** as a platform in your app _(if you have not added already)_.
+
+<Image
+  src="/static/books/helpshiftx/create_app_with_android_platform.png"
+  width="full"
+/>{" "}
+
+Click on configure button and then click on the _push notifications_ option.
+
+Enter your FCM key credentials per app, via the
+**Settings page > App listing in the left navigation > Scroll down to Push
+Notifications** settings section for the app.
+
+<Image src="/static/books/helpshiftx/add_fcm_token.png" width="full" />{" "}
+
+For FCM users, the API key can be found at your
+[Firebase API Console](https://console.firebase.google.com/)
+
+<Image src="/static/books/android/fcm_console.png" width="full" />{" "}
+
+<br />
+
+### Configure SDK {#configure-sdkx-android}
+
+1. Send the device registration id to Helpshift via the `HelpshiftCocosBridge.registerDeviceToken` API
+
+```java
+public class MyFirebaseMessagingService extends FirebaseMessagingService {
+  // ...
+  @Override
+  public void onNewToken(String newToken) {
+    // your logic to save the token here for the app
+    HelpshiftCocosBridge.registerPushToken(newToken);
+  }
+}
+```
+
+2. Using FCM's new `FirebaseMessagingService` which provides "RemoteMessage" argument, you should use `HelpshiftCocosBridge.handlePush` API which takes a Map argument.
+
+```java
+@Override
+public void onMessageReceived(RemoteMessage message) {
+  Map<String, String> data = message.getData();
+  String origin = data.get("origin");
+  if (origin != null && origin.equals("helpshift")) {
+      HelpshiftCocosBridge.handlePushNotification(getApplicationContext(), data);
```

Comment:
should we include install call here ?

Reply:
We do a `install` call with cached values inside `handlePushNotification` method.